bundle/deploy/state.go

```diff
@@ -0,0 +1,128 @@
+package deploy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/fileset"
+)
+
+const DeploymentStateFileName = "deployment-state.json"
+
+type File struct {
+	Absolute string `json:"absolute"`
+	Relative string `json:"relative"`
+}
+
+type Filelist []File
+
+type DeploymentState struct {
+	Version int64    `json:"version"`
```

Comment:
This is the deployment number, not the schema version of the deployment file. I think we need both; a number to identify the schema of the deployment state (to accommodate schema changes, we'd need to be able to upconvert from e.g. v1 to v2 internally), as well as a deployment sequence number.

Then we'd also protect against a deployment made by v2 from being mutated by a CLI on v1.